tests/jenkins/test_rest_server.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+rest_server_uri="$1"
+job_name="e2e-test-$RANDOM-$RANDOM"
+
+# get token
+token=$(curl -X POST -d "username=admin" -d "password=admin-password" -d "expiration=36000" ${rest_server_uri}/api/v1/authn/basic/login | jq -r ".token")
+if [ -z ${TOKEN} ]; then
+  echo "Cannot get token."
+  exit 1
+fi
+
+# submit job
+cat << EOF | curl -X POST -H "Authorization: Bearer ${token}" -H "Content-Type: application/json" -d @- ${rest_server_uri}/rest-server/api/v2/user/admin/jobs
+{
+  "jobName": "${job_name}",
+  "image": "docker.io/openpai/alpine:bash",
+  "taskRoles": [
+    {
+      "name": "test",
+      "taskNumber": 1,
+      "cpuNumber": 1,
+      "memoryMB": 2048,
+      "command": "/bin/bash --version"
+    }
+  ]
+}
+EOF
+
+# check job status
+while true; do
+  sleep 10s
+  status=$(curl ${rest_server_uri}/rest-server/api/v2/user/admin/jobs/${job_name} | jq -r ".jobStatus.state")
+  case $status in
+    "SUCCEEDED") break ;;
+    "WAITING"|"RUNNING") exit 1 ;;
+    *) ;;
+  esac
+done
```

Comment:
Do we have timeout for this?

Reply:
https://github.com/microsoft/pai/blob/132dfe281e41eb0c2b37401272cd9494e2a8c5e7/Jenkinsfile#L166-L169